### PR TITLE
Fix Issue-769

### PR DESCRIPTION
**Pull request by dataclouder**
This is a fix for Issue #769 "Plugin did not respond : terraform-provider-vcd may crash with Terraform 1.1+ on some OSes"

The VCD plugin was occasionally writing to Terraform `stdout` in order to transmit information to the user. Such use of `stdout` was undocumented, and the availability of the file descriptor that allowed such writing was finally removed in recent versions of Terraform CLI tool.

This fix removes every instance of using the above undocumented behavior, replacing it with the following:

* messages that were used as auxiliary list of items during `terraform import` are now formatted into an error message and displayed through a regular Terraform error, as they should have been in the first place.
* messages directed to the screen to communicate upload progress or suggest behavior are now written to the plugin log (`go-vcloud-director.log`) with a label (`[SCREEN]`) that makes it easy to find and eventually pipe to a visualization device. such as executing the command below in a separate terminal screen while `terraform apply` is running

```
  tail -f go-vcloud-director.log | grep '\[SCREEN\]'
```

### CODE REVIEW

File: vcd/config.go [hunk 1]
```diff
@@ -635,59 +633,42 @@ func buildUserAgent(version, sysOrg string) string {
...
 		// Its purpose is to alert the developer that there was an improper use of `dSet`
 		// The warning will work when testing with either `go test` or `make install` + `terraform apply`
-		fprintfNoErr(stdout,
-			"*** ERROR: only scalar values should be used for dSet() - detected '%s' (called from %s) \n",
-			reflect.TypeOf(value).Kind(), callFuncName())
-		fprintlnNoErr(stdout, starLine)
+		// ------------------------------------------------------------//
+		// NOTE FOR REVIEWERS: given that we lost the ability of using //
+		// Terraform stdout, is it OK to make this a panic instead?    //
+		// ------------------------------------------------------------//
```
Comment (vbauzysvmware) [inline vcd/config.go:661]: I would vote for panic, as we easily can miss log messages.

**Review by vbauzysvmware — APPROVED** [review]

**Review by Didainius — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
+## 3.5.1 (January 12, 2022)
+
+## BUG FIXES
+Fix Issue #769 "Plugin did not respond : terraform-provider-vcd may crash with Terraform 1.1+ on some OSes".
```
Comment (lvirbalas) [inline CHANGELOG.md:4]: Missing new line.

File: vcd/resource_vcd_catalog_item.go [hunk 3]
```diff
@@ -126,7 +124,7 @@ func resourceVcdCatalogItemCreate(d *schema.ResourceData, meta interface{}) erro
 				log.Printf("vCD Error importing new catalog item: %#v", err)
 				return fmt.Errorf("vCD Error importing new catalog item: %#v", err)
 			}
-			fprintNoErr(terraformStdout, "vcd_catalog_item."+itemName+": vCD import catalog item progress "+progress+"%\n")
+			logForScreen("vcd_catalog_item", fmt.Sprintf("vcd_catalog_item."+itemName+": vCD import catalog item progress "+progress+"%%\n"))
```
Comment (lvirbalas) [inline vcd/resource_vcd_catalog_item.go:127]:
```suggestion
			logForScreen("vcd_catalog_item", fmt.Sprintf("vcd_catalog_item."+itemName+": VCD import catalog item progress "+progress+"%%\n"))
```

**Review by lvirbalas — APPROVED** [review]
Just a few nits and LGTM!

**Pull request merged**